### ISSUE TITLE
Start building a recommender system

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-server#89]
Several of us have had the idea of building a recommender system on top of the data gathered by moves. To do this right is complicated, and requires integration with a number of different APIs.

However, it might be interesting to start working on a rudimentary recommender system for short trips so that we can understand the issues involved.

For example, trips that are < 1/2 mile can easily be walked, and trips that are < 3 miles can easily be biked. So a simple recommender system might be to say that car trips of those distances could be replaced by the equivalent walk and bike trips.

But there are some complications. If your 1/2 mile or 3 mile "trip" was actually a trip segment of a much larger trip that involved a ~ 15 mile drive, then it may not make as much sense. For example:
- somebody might drive 2 miles to a store and then drive 15 miles to work, the recommendation wouldn't make much sense because then you won't have your car at the store.
- On the other hand, if the store trip was out of the way of the drive to work, you could recommend that the store trip be done on a different day, when the user could have biked

And so on and so forth.

So basically, think through some issues for this very narrow and simple recommendation generation and try to build a system. Remember to think about evaluation as well!

Are you generating useful recommendations, or just annoying the user?

Shanthi has a very, very rudimentary visualization of this done, so you have some javascript to start working with.

**Comment by shanthistyle** [comment]
I, @neerajbaid, and @joshzarrabi would like to claim this one.

We were first thinking of displaying the most common trips. Instead, our recommender system could analyze the five most common trips and suggest an improvement as that could have highest impact for the user (i.e. if any of them are less than a mile, walk it). Another facet of the recommender system could be to use the Google API to find better routes at that time and recommend it when they are to take that trip again next time (or more simply, make the suggestion at the end of the day when they are confirming their trips)

**Comment by shankari** [comment]
@gaukes and @jeffdh5 just filed #97 (detect sub-optimal trips) which is similar.
@lechiang, @zacatac and @jimmydief just picked #88 (display common trips) which is also part of your suggestion.

It looks like three teams want to work on recommender systems for this week.

That's great! Let's spend some time on this Thursday deciding exactly what such a system should look like, divvy up the work into three separate projects and have the three teams work together to get it to work.

This is now the fun part, where we work together as a team to build projects that are much bigger than a small three person project.

Also, as I commented in #97, NREL also wants to build a recommender system :) Adding @karunmj to the discussion so that he can track our efforts and learn from them...
